crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -509,29 +517,60 @@ impl<'src> Parser<'src> {
         self.recovery_context = saved_context;
     }
 
-    fn parse_delimited_list<T>(
+    /// Parses a comma separated list of elements into a vector where each element
+    /// is parsed using the given `parse_element` function.
+    fn parse_comma_separated_list_into_vec<T>(
         &mut self,
-        kind: RecoveryContextKind,
+        recovery_context_kind: RecoveryContextKind,
         parse_element: impl Fn(&mut Parser<'src>) -> T,
         allow_trailing_comma: bool,
     ) -> Vec<T> {
-        let mut progress = ParserProgress::default();
         let mut elements = Vec::new();
+        self.parse_comma_separated_list(
+            recovery_context_kind,
+            |p| elements.push(parse_element(p)),
+            allow_trailing_comma,
+        );
+        elements
+    }
+
+    /// Parses a comma separated list of elements where each element is parsed
+    /// sing the given `parse_element` function.
+    ///
+    /// The difference between this function and `parse_comma_separated_list_into_vec`
+    /// is that this function does not return the parsed elements. Instead, it is the
+    /// caller's responsibility to handle the parsed elements. This is the reason
+    /// that the `parse_element` parameter is bound to [`FnMut`] instead of [`Fn`].
+    ///
+    /// If `allow_trailing_comma` is `true`, the function will allow a trailing
+    /// comma at the end of the list, otherwise it will add an error.
+    fn parse_comma_separated_list(
+        &mut self,
+        recovery_context_kind: RecoveryContextKind,
+        mut parse_element: impl FnMut(&mut Parser<'src>),
+        allow_trailing_comma: bool,
+    ) {
+        let mut progress = ParserProgress::default();
 
         let saved_context = self.recovery_context;
         self.recovery_context = self
             .recovery_context
-            .union(RecoveryContext::from_kind(kind));
+            .union(RecoveryContext::from_kind(recovery_context_kind));
 
         let mut trailing_comma_range: Option<TextRange> = None;
 
+        // TODO(dhruvmanila): I think this `loop` can be converted into a `while` loop
+        // similar to the one in the `parse_list` function above.
         loop {
```

Comment:
Yeah I think that's true by having a `while !self.at(TokenKind::EndOfFile) {}`. I'm just a big fan of `loop` :D

Reply:
Yeah, I like the `loop` version myself, just want consistency so that it's easier to understand both as they're similar :)